src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs

```diff
@@ -515,6 +516,16 @@ public static SyntaxTree ParseWithRoundTripCheck(string text, CSharpParseOptions
                 options = options.WithConcurrentBuild(false);
             }
 
+#if TEST_OPERATION
+            // Create a compilation for the purpose of verifying operation tree only,
+            // so this won't interfere with test.
+            var compilationForOperationWalking = CSharpCompilation.Create(
```

Comment:
What I meant is it would be good to not duplicate calculation of arguments. Perhaps extracting arguments that aren't already into locals?